File: frontend/profiles/papelaria/papelaria-period.ts
/**
 * Período de retirada/entrega do perfil papelaria (camada 8.15) — espelho 1:1 de
 * src/main/java/com/meada/profiles/papelaria/PapelariaPeriod.java.
 *
 * Pedido sob encomenda carrega o dia (pickup_or_delivery_date) + a FAIXA do dia (este enum).
 * O PapelariaPeriodParityTest (backend) garante que os ids aqui e no enum Java nunca divergem.
 * A CHECK de papelaria_orders.delivery_period trava os mesmos ids no banco. Clone do
 * padaria-period.ts.
 */
export const PAPELARIA_PERIODS = [
  { id: 'manha', label: 'Manhã (8h–12h)' },
  { id: 'tarde', label: 'Tarde (13h–18h)' },
] as const

export type PapelariaPeriod = (typeof PAPELARIA_PERIODS)[number]
export type PapelariaPeriodId = PapelariaPeriod['id']

/** Rótulo pt-BR de um período (fallback: o próprio id se desconhecido). */
export function periodLabel(id: string): string {
  return PAPELARIA_PERIODS.find((p) => p.id === id)?.label ?? id
}

File: src/main/java/com/meada/profiles/papelaria/PapelariaPeriod.java
package com.meada.profiles.papelaria;

import java.util.Arrays;
import java.util.List;
import java.util.Optional;

/**
 * Período de retirada/entrega do perfil papelaria (camada 8.15 / perfil papelaria) — MATERIALIZADO,
 * espelho 1:1 de {@code frontend/profiles/papelaria/papelaria-period.ts}. Clone de
 * {@link com.meada.profiles.padaria.PadariaPeriod} (camada 8.8). Quando o pedido tem item sob
 * encomenda, carrega o dia ({@code pickup_or_delivery_date}) + a FAIXA do dia (este enum). Duas faixas
 * hardcoded — não slot por minuto. O {@code PapelariaPeriodParityTest} garante a paridade Java↔TS; a
 * CHECK de {@code papelaria_orders.delivery_period} (migration 59) trava os mesmos ids no banco.
 */
public enum PapelariaPeriod {
    MANHA("manha", "Manhã (8h–12h)"),
    TARDE("tarde", "Tarde (13h–18h)");

    private final String id;
    private final String label;

    PapelariaPeriod(String id, String label) {
        this.id = id;
        this.label = label;
    }

    public String id() {
        return id;
    }

    public String label() {
        return label;
    }

    /** Resolve um período pelo id estável. Optional vazio se inválido/null. */
    public static Optional<PapelariaPeriod> fromId(String id) {
        if (id == null) {
            return Optional.empty();
        }
        return Arrays.stream(values()).filter(p -> p.id.equals(id)).findFirst();
    }

    /** Todos os períodos, na ordem de declaração. */
    public static List<PapelariaPeriod> allActive() {
        return List.of(values());
    }
}
